aepp/flowservice.py

```diff
@@ -461,21 +493,146 @@ def createFlow(self, obj: dict = None) -> dict:
                 requires following keys : name, flowSpec, sourceConnectionIds, targetConnectionIds, transformations, scheduleParams.
         """
         if obj is None:
-            raise Exception("Require a dictionary to create the flow")
-        if "name" not in obj.keys():
-            raise KeyError("missing 'name' parameter in the dictionary")
-        if "flowSpec" not in obj.keys():
-            raise KeyError("missing 'flowSpec' parameter in the dictionary")
-        if "sourceConnectionIds" not in obj.keys():
-            raise KeyError("missing 'sourceConnectionIds' parameter in the dictionary")
-        if "targetConnectionIds" not in obj.keys():
-            raise KeyError("missing 'targetConnectionIds' parameter in the dictionary")
+            if any(param is None for param in [name, source_connection_id, target_connection_id]):
+                raise KeyError("Require either obj or all of 'name', 'source_connection_id', 'target_connection_id'")
+            if schedule_frequency not in ("minute", "hour"):
+                raise ValueError("schedule frequency has to be either minute or hour")
+            obj = {
+                "name": name,
+                "flowSpec": {
+                    "id": flow_spec_id,
+                    "version": version
+                },
+                "sourceConnectionIds": [
+                    source_connection_id
+                ],
+                "targetConnectionIds": [
+                    target_connection_id
+                ],
+                "transformations": [],
+                "scheduleParams": {
+                    "startTime": schedule_start_time,
+                    "frequency": schedule_frequency,
+                    "interval": str(schedule_interval)
+                }
+            }
+            if transformation_mapping_id is not None:
```

Comment:
what does the transformation_mapping_id do?

Reply:
The Transformation Mapping is the Data Prep MapperSet Id. To map fields to XDM schema.